compiler/src/org.graalvm.compiler.truffle.compiler/src/org/graalvm/compiler/truffle/compiler/phases/inlining/CallNode.java

```diff
@@ -418,6 +426,7 @@ public void collectInlinedTargets(TruffleInliningData inliningPlan) {
 
     public enum State {
         Cutoff,
+        Excluded,
```

Comment:
I'm not a fan of adding an additional state for this. While it might seem useful to have the `Excluded` in the inlining trace I think that the state of the of the node and reason for that state are orthogonal issues. The policy should just check if the targets was excluded and decide not to inline it, leaving it's state as Explored (or potentially Cutoff if we decide to not even explore excluded nodes).

Reply:
Right now, if a node takes the excluded path (i.e. [gets continued without inlined](https://github.com/oracle/graal/pull/3524/files#diff-e7b7dbb1dab8bdbd05febd1974d7ee8c960438950b1f69932ea14df31bf0bc4bR120)) the status remains as `Expanded` and not `Explored` or `Cutoff`. Is that a status you'd expected? And am I to manually set it to something else?

Reply:
Sorry, a bit of a terminology mixup here, Expanded and Explored are the same thing, but I should have used Expanded since that's in the codebase. I think it's fine to keep it as `Expanded`, that just means that we did the partial evaluation of that call target but decided not to inline it.